onnxruntime/core/framework/session_state.cc

```diff
@@ -43,9 +43,14 @@ const SequentialExecutionPlan* SessionState::GetExecutionPlan() const {
   return p_seq_exec_plan_.get();
 }
 
-void SessionState::AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue) {
-  ORT_ENFORCE(mlvalue_index >= 0 && mlvalue_index <= mlvalue_name_idx_map_.MaxIdx());
-  initialized_tensors_.insert({mlvalue_index, mlvalue});
```

Comment:
The mlvalue index is coming from our internal code isn't it? As such we'd never expect it to be wrong and throwing is more appropriate?

Reply:
For me, I would think always returning a 'Status' is simpler. Otherwise, for each case I have to think whether I should throw or return an error. 

---
In reply to: [261360155](https://github.com/Microsoft/onnxruntime/pull/520#discussion_r261360155) [](ancestors = 261360155)

Reply:
I think of ORT_ENFORCE more like an assert, but you always want it to run run the check even in release builds. Generally I use it for validation of things that are purely internal to our code and not dependent on input from a user. Either achieve the same thing - we error out, so it's more a signal of whether the problem is purely internal or not.